src/System.Private.CoreLib/shared/System/Buffers/Binary/Reader.cs

```diff
@@ -105,10 +105,11 @@ public static uint ReverseEndianness(uint value)
             // Testing shows that throughput increases if the AND
             // is performed before the ROL / ROR.
 
-            uint mask_xx_zz = (value & 0x00FF00FFU);
-            uint mask_ww_yy = (value & 0xFF00FF00U);
-            return ((mask_xx_zz >> 8) | (mask_xx_zz << 24))
-                + ((mask_ww_yy << 8) | (mask_ww_yy >> 24));
+            uint mask_xx_zz = value & 0x00FF00FFU;
+            uint mask_ww_yy = value & 0xFF00FF00U;
+
+            return BitOps.RotateRight(mask_xx_zz, 8)
```

Comment:
The benefit here (as described in the summary) is better self-documentation, and code analysis, etc.
I expect perf would be the same, please advise if not